### PR TITLE
make migration error clear

### PR DESCRIPTION
**Pull request by 0xaptosj**
add 2 new error type for name already migrated and cannot migrate subdomain before migrate domain

### CODE REVIEW

File: router/sources/router.move [hunk 1]
```diff
@@ -363,6 +365,14 @@ module router::router {
             );
             let token_id = aptos_names::token_helper::latest_token_id(&tokendata_id);
 
+            // Domain must migrate before subdomain, throw error if this is a subdomain but domain has not been migrated
+            if (option::is_some(&subdomain_name)) {
+                assert!(
+                    exists_in_v2(domain_name, option::none()) && is_name_owner(user_addr, domain_name, option::none()),
```
Comment (rtso) [inline router/sources/router.move:371]: I think this error would apply even if the user is not owner of the domain. Do we need the `is_name_owner` check?

Comment (0xaptosj) [reply router/sources/router.move:371]: i think user must be the domain owner to be able to migrate the subdomain, @angieyth @BriungRi i think so, right?

Comment (angieyth) [reply router/sources/router.move:371]: I  think Renee means we should throw the error when the domain is not v2 (whether the user is the owner or not)

Comment (angieyth) [reply router/sources/router.move:371]: but also I think you don't have to be the domain owner to migrate the subdomain(?) you need to be the subdomain owner to do so.

Comment (rtso) [reply router/sources/router.move:371]: +1 to @angieyth, for example I created a domain and subdomain, then transferred the subdomain to someone else. They should have the power to migrate the subdomain.

**Review by angieyth — APPROVED** [review]

File: router/sources/router.move (hunk 1 above)
Comment (0xaptosj) [reply router/sources/router.move]: make sense, updated

**Pull request merged**